src/ray/core_worker/task_manager.h

```diff
@@ -616,14 +615,14 @@ class TaskManager : public TaskFinisherInterface, public TaskResubmissionInterfa
 
  private:
   struct TaskEntry {
-    TaskEntry(const TaskSpecification &spec_arg,
+    TaskEntry(TaskSpecification spec_arg,
               int num_retries_left_arg,
               size_t num_returns,
               TaskStatusCounter &counter,
               int64_t num_oom_retries_left)
-        : spec(spec_arg),
+        : spec(std::move(spec_arg)),
           num_retries_left(num_retries_left_arg),
-          counter(counter),
+          counter(&counter),
```

Comment:
Bad naming... Avoid naming ctor argument and data member the same.

Reply:
can fix in follow-up want to minimize more changes here